include/zephyr/net/prometheus/pm_collector.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+
+#include <zephyr/net/prometheus/pm_metric.h>
+
+#include <stddef.h>
+
+struct prometheus_collector_node {
+	struct prometheus_metric *metric;
+	struct pm_collector_node *next;
```

Comment:
Instead of our own list pointers, would it make sense to use the slist API here, WDYT?

Reply:
I wanted to use slist apis at first. I will update with slist instead of my linked list implementation but I should be try. We should will measure overhead between implementations.